Model/Resource.php

```diff
@@ -199,4 +202,13 @@ public function getBoolValue()
     {
         return $this->boolValue;
     }
+
+    /**
+     * @param DateTime $value
+     * @return DateTimeImmutable
+     */
+    private function convertDateToImmutable(DateTime $value)
```

Comment:
I would use `DateTimeInterface` as typehint because it would be compatible with doctrine's immutable types